src/hooks/BookAppointment/useAppointmentCalendar.hook.tsx

```diff
@@ -1,37 +1,181 @@
-  const [month, setMonth] = useState(nextMonth);
-  const currentStyle = { backgroundColor: '#4579EE' };
+  const [month, setMonth] = useState<Date>(nextMonth);
+  const [showCalendar, setShowCalendar] = useState<boolean>(false);
+  const [freeSlots, setFreeSlots] = useState<FreeSlotProps[]>([]);
+  const currentStyle = { backgroundColor: `${BLUE}` };
+  const bookedStyle = { color: `${HINT}` };
+  const { data } = appointmentApi.useGetSpecializationByIdQuery(specialization);
+
+  useEffect(() => {
+    if (specialization !== undefined && data) {
+      setFreeSlots(data);
+    }
+  }, [specialization, data]);
+
+  useEffect(() => {
+    if (freeSlots.length === 0) {
+      setSelectedDay(null);
+    }
+  }, [freeSlots]);
+
+  // work with calendar
+  const ThreeMothPeriod = [];
+  for (let i = 0; i < 3; i++) {
+    const year = today.getFullYear();
+    const month = today.getMonth() + i;
+    const daysInMonth = new Date(year, month + 1, 0).getDate();
+
+    for (let day = 1; day <= daysInMonth; day++) {
+      const date = new Date(year, month, day);
+      date.setHours(0, 0, 0, 0);
+      ThreeMothPeriod.push(date);
+    }
+  }
+
+  // Set all time to 0 and receive the number equivalent by getTime() function
+  const formattedThreeMothPeriod = ThreeMothPeriod.map((date) => {
+    date.setHours(0, 0, 0, 0);
+    return date.getTime();
+  });
+
+  const formattedFreeSlots = freeSlots.map((item) => {
+    const date = new Date(item.start);
+    date.setHours(0, 0, 0, 0);
+    return date.getTime();
+  });
+
+  const allBookedDates = formattedThreeMothPeriod.filter(
+    (date) => !formattedFreeSlots.includes(date)
+  );
 
-  const [showCalendar, setShowCalendar] = useState(false);
+  //array with not avalible dates
+  const formattedBookedDates = allBookedDates.map((date) => {
+    const d = new Date(date);
+    const year = d.getFullYear();
+    const month = d.getMonth();
+    const day = d.getDate();
+    return new Date(year, month, day);
+  });
 
-  // Todo :  hardcode, just an example till connection with backend
   const isDisabled = (day: Date) => {
     return day > new Date();
   };
 
   const bookedDays = [
-    new Date(2023, 4, 10),
-    new Date(2023, 4, 12),
-    new Date(2023, 5, 20),
+    ...formattedBookedDates,
     { before: new Date(), modifiers: { disabled: isDisabled } },
   ];
 
-  // Todo: styles for days without empty slots. not sure if we need them at this moment, maybe in next sprint will be deleted or moved to styles
-  const bookedStyle = { color: '#808080' };
+  const selectedDate = new Date(selectedDay);
+
+  const filteredSlots = freeSlots.filter((slot) => {
+    const slotDate = new Date(slot.start);
+    return (
+      slotDate.getDate() === selectedDate.getDate() &&
+      slotDate.getMonth() === selectedDate.getMonth() &&
+      slotDate.getFullYear() === selectedDate.getFullYear()
+    );
+  });
+
+  // Sort all slots by time and leave only unique slots
+  filteredSlots.sort((a, b) => {
```

Comment:
nono no find other solution, it will recalculate with each rerender

Reply:
fix.